airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java

```diff
@@ -227,17 +232,141 @@ private int insertConfigRecord(DSLContext ctx, OffsetDateTime timestamp, String
+        .where(AIRBYTE_CONFIGS.CONFIG_TYPE.eq(configType), AIRBYTE_CONFIGS.CONFIG_ID.eq(configId))
+        .execute();
+  }
+
+  @VisibleForTesting
+  void copyConfigsFromSeed(DSLContext ctx, ConfigPersistence seedConfigPersistence) throws SQLException {
+    LOGGER.info("Loading data to config database...");
+
+    Map<String, Stream<JsonNode>> seedConfigs;
+    try {
+      seedConfigs = seedConfigPersistence.dumpConfigs();
+    } catch (IOException e) {
+      throw new SQLException(e);
+    }
+
+    OffsetDateTime timestamp = OffsetDateTime.now();
+    int insertionCount = seedConfigs.entrySet().stream().map(entry -> {
+      String configType = entry.getKey();
+      return entry.getValue().map(configJson -> {
+        String idFieldName = ConfigSchemaMigrationSupport.CONFIG_SCHEMA_ID_FIELD_NAMES.get(configType);
+        return insertConfigRecord(ctx, timestamp, configType, configJson, idFieldName);
+      }).reduce(0, Integer::sum);
+    }).reduce(0, Integer::sum);
+
+    LOGGER.info("Config database data loading completed with {} records", insertionCount);
+  }
+
+  @VisibleForTesting
+  void updateConfigsFromSeed(DSLContext ctx, ConfigPersistence seedConfigPersistence) throws SQLException {
+    LOGGER.info("Config database has been initialized; updating connector definitions from the seed if necessary...");
+
+    try {
+      Set<String> usedConnectorRepos = getUsedConnectorRepos(ctx);
+      Map<String, Record2<String, String>> currentRepoToIdAndVersions = getCurrentRepoToIdAndVersions(ctx);
+
+      OffsetDateTime timestamp = OffsetDateTime.now();
+      int insertionCount = 0;
+      int updatedCount = 0;
+
+      List<StandardSourceDefinition> latestSources = seedConfigPersistence.listConfigs(
+          ConfigSchema.STANDARD_SOURCE_DEFINITION, StandardSourceDefinition.class);
+      Record2<Integer, Integer> sourceUpdate = updateConnectorDefinitions(ctx, timestamp, ConfigSchema.STANDARD_SOURCE_DEFINITION,
+          latestSources, usedConnectorRepos, currentRepoToIdAndVersions);
+      insertionCount += sourceUpdate.value1();
+      updatedCount += sourceUpdate.value2();
+
+      List<StandardDestinationDefinition> latestDestinations = seedConfigPersistence.listConfigs(
+          ConfigSchema.STANDARD_DESTINATION_DEFINITION, StandardDestinationDefinition.class);
+      Record2<Integer, Integer> destinationUpdate = updateConnectorDefinitions(ctx, timestamp, ConfigSchema.STANDARD_DESTINATION_DEFINITION,
+          latestDestinations, usedConnectorRepos, currentRepoToIdAndVersions);
+      insertionCount += destinationUpdate.value1();
+      updatedCount += destinationUpdate.value2();
+
+      LOGGER.info("Connector definitions have been updated ({} new connectors, and {} updates)", insertionCount, updatedCount);
+    } catch (IOException | JsonValidationException e) {
+      throw new SQLException(e);
+    }
+  }
+
+  private <T> Record2<Integer, Integer> updateConnectorDefinitions(DSLContext ctx,
```

Comment:
nit: short comment here on the return being <insertionCount, updatedCount>

Reply:
Will replace it with a value wrapper class for clarity. `Record2` is both confusing and verbose to construct.